src/main/java/ru/practicum/shareit/item/service/ItemServiceImplementation.java

```diff
@@ -18,57 +31,127 @@
+        }
+        return itemWithBookingDatesDtoList;
     }
 
     @Override
     public List<ItemDto> searchItem(String text) {
         if (text.isBlank()) {
             return new ArrayList<>();
         }
-        return itemStorage.searchItem(text).stream()
+        return itemRepository.searchItem(text).stream()
                 .map((itemMapper::mapItemToItemDto)).collect(Collectors.toList());
     }
 
-    private void checkIfUserIdExists(int id) {
-        if (!itemStorage.checkIfUserIdAlreadyExists(id)) {
-            throw new ObjectNotFoundException(String.format("User with id: %d is not existed", id));
-        }
+    @Override
+    public CommentDto addComment(CommentDto commentDto, int itemId, int userId) {
+        User user = getUserById(userId);
+        Item item = getItemById(itemId);
+        bookingRepository.checkIfUserBookedItem(userId, itemId)
+                .orElseThrow(() -> new IncorrectInputException(String
+                        .format("User ID: %d did not book item ID: %d", userId, itemId)));
+        Comment comment = commentMapper.mapCommentDtoToComment(commentDto);
+        comment.setItem(item);
+        comment.setCommentAuthor(user);
+        comment.setCreated(LocalDateTime.now());
+        return commentMapper.mapCommentToCommentDto(commentRepository.save(comment));
     }
 
-    private void checkItemOwner(int userId, int itemId) {
-        if (!itemStorage.checkItemOwner(userId, itemId)) {
-            throw new ObjectNotFoundException("Item does not belong to User and can not be updated!");
-        }
+    private User getUserById(int userId) {
+        return userRepository.findById(userId)
+                .orElseThrow(() -> new ObjectNotFoundException(String.format("User Id %d is not existed", userId)));
     }
 
-    private void checkIfItemIdExists(int itemId) {
-        if (!itemStorage.checkIfItemIdExists(itemId)) {
-            throw new ObjectNotFoundException(String.format("Item with ID %d was not found", itemId));
-        }
+    private Item getItemById(int itemId) {
+        return itemRepository.findById(itemId)
+                .orElseThrow(() -> new ObjectNotFoundException(String.format("Item Id %d is not existed", itemId)));
+    }
+
+    private BookingOwnerDTO getLastBooking(int itemId) {
+        return bookingMapper
+                .mapBookingToBookingOwnerDTO(bookingRepository.searchLatestBooking(itemId));
+    }
+
+    private BookingOwnerDTO getNextBooking(int itemId) {
+        return bookingMapper.mapBookingToBookingOwnerDTO(bookingRepository.searchNearestBooking(itemId));
+    }
+
+    private boolean checkIfItemOwnerEqualsUserId(Item item, int userId) {
+        return item.getOwner().getId() == userId;
+    }
+
+    private List<CommentDto> getListOfComments(Item item) {
+        return commentRepository.getCommentsOfItem(item.getId())
+                .stream().map(commentMapper::mapCommentToCommentDto).collect(Collectors.toList());
```

Comment:
Лучше бить методы стрима по строкам для лучшей читаемости:
```suggestion
                .stream()
                .map(commentMapper::mapCommentToCommentDto)
                .collect(Collectors.toList());
```

Reply:
Исправлено